coderd/tailnet.go

```diff
@@ -103,14 +103,7 @@ func NewServerTailnet(
 		transport:            tailnetTransport.Clone(),
 	}
 	tn.transport.DialContext = tn.dialContext
-
-	// Bugfix: for some reason all calls to tn.dialContext come from
-	// "localhost", causing connections to be cached and requests to go to the
-	// wrong workspaces. This disables keepalives for now until the root cause
-	// can be found.
-	tn.transport.MaxIdleConnsPerHost = -1
-	tn.transport.DisableKeepAlives = true
-
+	tn.transport.MaxIdleConnsPerHost = 10
```

Comment:
why did we override the default of 2 per host and set an unlimited number of total idle connections?

I feel like this deserves a comment